app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala

```diff
@@ -1086,6 +1086,70 @@ object ConsoleCli {
                 case other => other
               }))
         ),
+      cmd("createnumericevent")
+        .action((_, conf) =>
+          conf.copy(command = CreateNumericEvent(eventName = "",
+                                                 maturationTime = Instant.MIN,
+                                                 maxValue = 0,
+                                                 isSigned = false,
+                                                 unit = "",
+                                                 precision = 0)))
+        .text("Registers an oracle event that uses digit decomposition when signing the number")
+        .children(
+          arg[String]("name")
+            .text("Name for this event")
+            .required()
+            .action((name, conf) =>
+              conf.copy(command = conf.command match {
+                case createNumericEvent: CreateNumericEvent =>
+                  createNumericEvent.copy(eventName = name)
+                case other => other
+              })),
+          arg[Instant]("maturationtime")
+            .text("The earliest expected time an outcome will be signed, given in epoch second")
+            .required()
+            .action((time, conf) =>
+              conf.copy(command = conf.command match {
+                case createNumericEvent: CreateNumericEvent =>
+                  createNumericEvent.copy(maturationTime = time)
+                case other => other
+              })),
+          arg[Long]("maxvalue")
+            .text("Maximum value of this event")
+            .required()
+            .action((max, conf) =>
+              conf.copy(command = conf.command match {
+                case createNumericEvent: CreateNumericEvent =>
+                  createNumericEvent.copy(maxValue = max)
+                case other => other
+              })),
+          opt[Unit]("signed")
+            .text("Whether the outcomes can be negative")
+            .action((_, conf) =>
+              conf.copy(command = conf.command match {
+                case createNumericEvent: CreateNumericEvent =>
+                  createNumericEvent.copy(isSigned = true)
+                case other => other
+              })),
+          arg[String]("unit")
```

Comment:
This is more of a general comment that might be worth pursuing in a future PR, I think it might be possible to factor out some of these common args into defs so that when they are used multiple places you only need to pass a function putting the value into the case class via copy

Reply:
Worth filing an issue for this?